### PR TITLE
HHH-15406 Tests annotated with org.hibernate.testing.RequiresDialect having as value a deprecated dialect are not executed

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15406
https://hibernate.atlassian.net/browse/HHH-15413
https://hibernate.atlassian.net/browse/HHH-15414

### CODE REVIEW

File: documentation/src/test/java/org/hibernate/userguide/inheritance/SingleTableDiscriminatorFormulaTest.java [hunk 1]
```diff
@@ -15,32 +15,30 @@
...
-				CreditAccount.class,
-		};
-	}
+@Jpa(
+		annotatedClasses = {
+				SingleTableDiscriminatorFormulaTest.DebitAccount.class,
+				SingleTableDiscriminatorFormulaTest.CreditAccount.class,
+		}
+)
+public class SingleTableDiscriminatorFormulaTest {
```
Comment (jrenaat) [inline documentation/src/test/java/org/hibernate/userguide/inheritance/SingleTableDiscriminatorFormulaTest.java:37]: Do we need an after-the-test-cleanup method ?

**Review by jrenaat — COMMENTED** [review]
LGTM
I saw that some of the tests that persist entities don't have the typical cleanup method; not sure if we add that by default or not, leaving that to your criterium.

**Comment by dreab8** [comment]
@jrenaat depends in a test class like [SingleTableDiscriminatorFormulaTest](https://github.com/hibernate/hibernate-orm/pull/5177/files#diff-9328cd452a62d66d9ec7e2cd2e959027bc40b235db49df8972269a66840e485f ) that contains only one test  it is not a problem because at the end we are dropping the schema,  in a test like [ExplicitLockingTest](https://github.com/hibernate/hibernate-orm/pull/5177/files#diff-c48cd7fa11568202a72f9d10ae22b46513c11485faa99c927358b22bd028979d) that contains more than one test cleaning up the garbage leaved by the execution of each test is important so I added a missing clean up method.

File: documentation/src/test/java/org/hibernate/userguide/sql/MySQLStoredProcedureTest.java [hunk 2]
```diff
@@ -8,256 +8,262 @@
...
-    @Override
-    protected Class<?>[] getAnnotatedClasses() {
-        return new Class<?>[] {
-                Person.class,
-                Partner.class,
-                Phone.class,
-                Call.class,
-        };
-    }
+@RequiresDialect(value = MySQLDialect.class, majorVersion = 5)
```
Comment (beikov) [inline documentation/src/test/java/org/hibernate/userguide/sql/MySQLStoredProcedureTest.java:43]: You need to add `@SkipForDialect(value = TiDBDialect.class, comment = "TiDB doesn't support stored procedures")`

Comment (dreab8) [reply documentation/src/test/java/org/hibernate/userguide/sql/MySQLStoredProcedureTest.java:45]: Thanks @beikov done :)

**Review by beikov — APPROVED** [review]

**Comment by jrenaat** [comment]
> @jrenaat depends in a test class like [SingleTableDiscriminatorFormulaTest](https://github.com/hibernate/hibernate-orm/pull/5177/files#diff-9328cd452a62d66d9ec7e2cd2e959027bc40b235db49df8972269a66840e485f) that contains only one test it is not a problem because at the end we are dropping the schema, in a test like [ExplicitLockingTest](https://github.com/hibernate/hibernate-orm/pull/5177/files#diff-c48cd7fa11568202a72f9d10ae22b46513c11485faa99c927358b22bd028979d) that contains more than one test cleaning up the garbage leaved by the execution of each test is important so I added a missing clean up method.

Excellent, thanks

**Pull request merged**